### PR TITLE
[dplay] add support for it.dplay.com and dplay.dk

### PR DESCRIPTION
**Pull request by aidan-**
- Updated DPlay extractor to include support for videos from [it.dplay.com](http://it.dplay.com) and [www.dplay.dk](http://www.dplay.dk)
- Remove an unnecessary additional JSON request from original extractor

Let me know if anything needs to be changed.

### CODE REVIEW

File: youtube_dl/extractor/dplay.py [hunk 1]
```diff
@@ -2,50 +2,85 @@
...
+                'season_number': 6,
+                'episode_number': 12,
+            },
+        }
+    ]
 
     def _real_extract(self, url):
+        #this extrator works with it.dplay.com, www.dplay.se and www.dplay.dk
+        # so we need to determine the domain to send the requests to
+        domain = self._search_regex('^([a-z\.\:\/]+(com|dk|se))', url, 'domain')
```
Comment (dstftw) [inline youtube_dl/extractor/dplay.py]: This should be extracted right from `_VALID_URL`.

File: youtube_dl/extractor/dplay.py [hunk 2]
```diff
@@ -2,50 +2,85 @@
...
-
-    _TEST = {
-        'url': 'http://www.dplay.se/nugammalt-77-handelser-som-format-sverige/season-1-svensken-lar-sig-njuta-av-livet/',
-        'info_dict': {
-            'id': '3172',
-            'ext': 'mp4',
-            'display_id': 'season-1-svensken-lar-sig-njuta-av-livet',
-            'title': 'Svensken lär sig njuta av livet',
-            'duration': 2650,
+    _VALID_URL = r'http://(it|www)\.dplay\.(com|dk|se)/[^/]+/(?P<id>[^/?#]+)'
```
Comment (dstftw) [inline youtube_dl/extractor/dplay.py]: Do not capture groups if you are not going to use them.

File: youtube_dl/extractor/dplay.py [hunk 3]
```diff
@@ -2,50 +2,85 @@
...
-            video_id, 'Getting manifest url for hls stream')['hls']
+        manifest_url = info['hls']
         formats = self._extract_m3u8_formats(
             manifest_url, video_id, ext='mp4', entry_protocol='m3u8_native')
 
         return {
             'id': video_id,
             'display_id': display_id,
-            'title': info['title'],
+            'title': info.get('title'),
```
Comment (dstftw) [inline youtube_dl/extractor/dplay.py]: Title should be mandatory.

File: youtube_dl/extractor/dplay.py [hunk 4]
```diff
@@ -2,50 +2,85 @@
...
+        domain = self._search_regex('^([a-z\.\:\/]+(com|dk|se))', url, 'domain')
+
         display_id = self._match_id(url)
         webpage = self._download_webpage(url, display_id)
         video_id = self._search_regex(
             r'data-video-id="(\d+)"', webpage, 'video id')
 
-        info = self._download_json(
-            'http://www.dplay.se/api/v2/ajax/videos?video_id=' + video_id,
+        video_url = urljoin(domain, 'api/v2/ajax/videos?video_id=')
```
Comment (dstftw) [inline youtube_dl/extractor/dplay.py]: Use `compat_urlparse.urljoin` instead.

**Comment by aidan-** [comment]
Thanks for the feedback.  I have addressed each of the issues.

File: youtube_dl/extractor/dplay.py [hunk 5]
```diff
@@ -5,40 +5,70 @@
...
 from .common import InfoExtractor
 from ..utils import int_or_none
-
+from ..compat import compat_urlparse
 
 class DPlayIE(InfoExtractor):
-    _VALID_URL = r'http://www\.dplay\.se/[^/]+/(?P<id>[^/?#]+)'
-
-    _TEST = {
-        'url': 'http://www.dplay.se/nugammalt-77-handelser-som-format-sverige/season-1-svensken-lar-sig-njuta-av-livet/',
```
Comment (dstftw) [inline youtube_dl/extractor/dplay.py]: Why did you remove this test? It does not work with your changes.

Comment (aidan-) [inline youtube_dl/extractor/dplay.py]: The _TEST variable has been changed to an array and two additional tests have been added?  The documentation I read mentioned to do that in the instance of more than one test case.

Comment (dstftw) [inline youtube_dl/extractor/dplay.py]: Didn't notice you placed it second. Anyway it fails after this changes:
```
> py26yt -v test:DPlay_1
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'test:DPlay_1']
[debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
[debug] youtube-dl version 2016.02.22
[debug] Git HEAD: 73ab2dd
[debug] Python version 2.6.6 - Windows-2003Server-5.2.3790-SP2
[debug] exe versions: avconv 11.3, avprobe 11.3, ffmpeg N-77883-gd7c75a5, ffprobe N-77883-gd7c75a5, rtmpdump 2.4
[debug] Proxy map: {}
[TestURL] Test URL: http://www.dplay.se/nugammalt-77-handelser-som-format-sverige/season-1-svensken-lar-sig-njuta-av-livet/
[DPlay] season-1-svensken-lar-sig-njuta-av-livet: Downloading webpage
[DPlay] 3172: Downloading JSON metadata
[DPlay] 3172: Downloading m3u8 information
ERROR: Failed to download m3u8 information: HTTP Error 403: Forbidden (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you
 are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete outp
ut.
  File "C:\Dev\git\youtube-dl\master\youtube_dl\extractor\common.py", line 355, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "C:\Dev\git\youtube-dl\master\youtube_dl\YoutubeDL.py", line 1905, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "C:\python\python266\lib\urllib2.py", line 397, in open
    response = meth(req, response)
  File "C:\python\python266\lib\urllib2.py", line 510, in http_response
    'http', request, response, code, msg, hdrs)
  File "C:\python\python266\lib\urllib2.py", line 435, in error
    return self._call_chain(*args)
  File "C:\python\python266\lib\urllib2.py", line 369, in _call_chain
    result = func(*args)
  File "C:\python\python266\lib\urllib2.py", line 518, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
```
Before this change:
```
> py26yt -v test:DPlay
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'test:DPlay']
[debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
[debug] youtube-dl version 2016.02.22
[debug] Git HEAD: 73ab2dd
[debug] Python version 2.6.6 - Windows-2003Server-5.2.3790-SP2
[debug] exe versions: avconv 11.3, avprobe 11.3, ffmpeg N-77883-gd7c75a5, ffprobe N-77883-gd7c75a5, rtmpdump 2.4
[debug] Proxy map: {}
[TestURL] Test URL: http://www.dplay.se/nugammalt-77-handelser-som-format-sverige/season-1-svensken-lar-sig-njuta-av-livet/
[DPlay] season-1-svensken-lar-sig-njuta-av-livet: Downloading webpage
[DPlay] 3172: Downloading JSON metadata
[DPlay] 3172: Getting manifest url for hls stream
[DPlay] 3172: Downloading m3u8 information
[debug] Invoking downloader on u'http://sbsvod02-vh.akamaihd.net/i/delivery/9b/64/9b647144-0347-4a96-9c91-255f82d7bde9/SE304349146527_,25,2,6,10,15,00k.mp4.csmi
l/index_0_av.m3u8?null='
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 265
[download] Destination: Svensken lr sig njuta av livet-3172.mp4
[download]   1.9% of ~888.61MiB at Unknown speed ETA Unknown ETA
ERROR: Interrupted by user
```

Comment (dstftw) [inline youtube_dl/extractor/dplay.py]: DPlay_2 fails as well.

Comment (aidan-) [inline youtube_dl/extractor/dplay.py]: Strange, It was definitely working 15days ago.  I'll take a look at this later today.  It could be due to the setting of the cookie that I removed.

**Comment by aidan-** [comment]
The latest commit fixes DPlay_1 and DPlay_2.  Looks like the cookie needs to be set (now?).

File: youtube_dl/extractor/dplay.py [hunk 6]
```diff
@@ -5,40 +5,91 @@
...
+                'https://secure.dplay.se/secure/api/v2/user/authorization/stream/%s?stream_type=hls' % video_id,
+                video_id, 'Getting manifest url for hls stream')['hls']
+        elif domain_tld == "dk":
+            self._set_cookie(
+                'secure.dplay.dk', 'dsc-geo',
+                '{"countryCode":"DK","expiry":%d}' % ((time.time() + 20 * 60) * 1000))
+
+            manifest_url = self._download_json(
+                'https://secure.dplay.dk/secure/api/v2/user/authorization/stream/%s?stream_type=hls' % video_id,
+                video_id, 'Getting manifest url for hls stream')['hls']
```
Comment (dstftw) [inline youtube_dl/extractor/dplay.py]: Code duplication should be eliminated.

**Comment by aidan-** [comment]
Removed cooking setting code.

**Comment by dstftw** [comment]
Can you squash commits?

File: youtube_dl/extractor/dplay.py [hunk 7]
```diff
@@ -5,40 +5,83 @@
...
-            'secure.dplay.se', 'dsc-geo',
-            '{"countryCode":"NL","expiry":%d}' % ((time.time() + 20 * 60) * 1000))
         # TODO: consider adding support for 'stream_type=hds', it seems to
         # require setting some cookies
-        manifest_url = self._download_json(
-            'https://secure.dplay.se/secure/api/v2/user/authorization/stream/%s?stream_type=hls' % video_id,
-            video_id, 'Getting manifest url for hls stream')['hls']
+        # get url's TLD to determine which cookie and url to use
+        domain_tld = domain.split(".")[-1:][0]
+        if domain_tld == "se" or domain_tld == "dk":
```
Comment (dstftw) [inline youtube_dl/extractor/dplay.py]: Use single quotes consistently.

File: youtube_dl/extractor/dplay.py [hunk 8]
```diff
@@ -5,40 +5,83 @@
...
-        self._set_cookie(
-            'secure.dplay.se', 'dsc-geo',
-            '{"countryCode":"NL","expiry":%d}' % ((time.time() + 20 * 60) * 1000))
         # TODO: consider adding support for 'stream_type=hds', it seems to
         # require setting some cookies
-        manifest_url = self._download_json(
-            'https://secure.dplay.se/secure/api/v2/user/authorization/stream/%s?stream_type=hls' % video_id,
-            video_id, 'Getting manifest url for hls stream')['hls']
+        # get url's TLD to determine which cookie and url to use
+        domain_tld = domain.split(".")[-1:][0]
```
Comment (dstftw) [inline youtube_dl/extractor/dplay.py]: Just take the last character `domain.split('.')[-1]`.

**Comment by aidan-** [comment]
All commits have been squashed down to one, and the use of single quotes is consistent.

**Comment by dstftw** [comment]
Cherry-picked in 950505370416e79073acdeaa73a1023950363266, thanks.
